### PR TITLE
Add user store rest api

### PR DESCRIPTION
**Pull request by Kanapriya**
## Purpose
Adding the implementation to https://github.com/wso2/product-is/issues/6813
Refer - https://github.com/wso2/product-is/issues/6832

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.user.store.configuration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-web</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/pom.xml]: new line

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * Contains all the user store related constants.
+ */
+public class UserStoreConstants {
+
+    private static final String SECONDARY_USERSTORE_PREFIX = "SUS-";
+    public static final String OPERATION_REPLACE = "REPLACE";
+    public static final String OPERATION_ADD = "ADD";
+    public static final String USER_STORE_PATH_COMPONENT = "/userstores";
+
+
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java]: unnecessary new line

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+            if (isUserStoreDomainIdFound(base64DecodeId(domainId))) {
+                userStoreConfigService.updateUserStoreByDomainName(base64DecodeId(domainId),
+                        createUserStorePutDTO(userStorePutReq));
+                UserStoreResponse userStoreResponseDTO = new UserStoreResponse();
+                userStoreResponseDTO.setId((base64EncodeId(userStorePutReq.getName())));
+                userStoreResponseDTO.setName(userStorePutReq.getName());
+                userStoreResponseDTO.setTypeId(base64EncodeId(userStorePutReq.getTypeName()));
+                userStoreResponseDTO.setTypeName(userStorePutReq.getTypeName());
+                userStoreResponseDTO.setDescription(userStorePutReq.getDescription());
+                userStoreResponseDTO.setProperties(buildUserStorePropertiesRes(null, userStorePutReq));
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java]: Seems to be this part is duplicated.

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        ERROR_CODE_FILTERING_NOT_IMPLEMENTED("50006",
+                "Filtering not supported.",
+                "Filtering capability is not supported in this version of the API."),
+        ERROR_CODE_SORTING_NOT_IMPLEMENTED("50007",
+                "Sorting not supported.",
+                "Sorting capability is not supported in this version of the API."),
+        ERROR_CODE_DOMAIN_ID_NOT_FOUND("50008",
+                "Resource not found.",
+                "Unable to find any user store's id with the provided identifier %s."),
+        ERROR_CODE_DATASOURCE_CONNECTION("500011",
```
Comment (Abilashini) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java]: error code has 6 digits. Should be 50011

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                "Unable to check RDBMS connection Health",
+                "Server Encountered an error while checking the data source connection."),
+        ERROR_CODE_RETRIEVING_USER_STORE_TYPE("500012",
+                "Unable to retrieve the user store implementations",
+                "Server Encountered an error while retrieving the implementations."),
+        ;
```
Comment (Abilashini) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java]: Please remove the comma at the end of the previous line and add this semicolon there

**Pull request merged**